node: don't consider internal masquerade addresses as node IP addresses

dcbw opened the pull request:
`I0123 00:45:01.424261 946716 node_ip_handler_linux.go:268] Node address annotation being set to: map[192.168.18.12:{} 2600:52:7:18::12:{} fd69::2:{}] addrChanged: true`

ovn-kube shouldn't be considering the fd69::2 internal masquerade address (or its IPv4 equiavlent) as a node IP address. It gets set by setNodeMasqueradeIPOnExtBridge().

ricky-rav commented:
Nice catch! I did the same check here: https://github.com/ovn-org/ovn-kubernetes/commit/d2ca5d6c32ad6ca1e7de4f9a9ae91598a36bd738# 
Would it make sense to reuse `isAddressReservedForInternalUse`?

dcbw commented:
> Would it make sense to reuse `isAddressReservedForInternalUse`?

@ricky-rav it absolutely does. Updated, and added testcases to verify basic behavior.

ricky-rav commented:
/LGTM
Thanks, @dcbw!
